### ISSUE TITLE
sensor values containing strings cause calculations to fail

### ISSUE THREAD

**Issue by MrBurnst** [jeroenterheerdt/HAsmartirrigation#402]
### What happened?

this is already mentioned in issue https://github.com/jeroenterheerdt/HAsmartirrigation/issues/395#issuecomment-2156325582 - sensor values can contain strings, which leads to the following error:

```
Logger: aiohttp.server
Source: /usr/local/lib/python3.12/site-packages/aiohttp/web_protocol.py:421
First occurred: 8:54:30 PM (5 occurrences)
Last logged: 8:55:08 PM

Error handling request
Traceback (most recent call last):
  File "/usr/local/lib/python3.12/statistics.py", line 327, in _exact_ratio
    return (x.numerator, x.denominator)
            ^^^^^^^^^^^
AttributeError: 'str' object has no attribute 'numerator'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.12/site-packages/aiohttp/web_protocol.py", line 452, in _handle_request
    resp = await request_handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/aiohttp/web_app.py", line 543, in _handle
    resp = await handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/aiohttp/web_middlewares.py", line 114, in impl
    return await handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/http/security_filter.py", line 92, in security_filter_middleware
    return await handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/http/forwarded.py", line 83, in forwarded_middleware
    return await handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/http/request_context.py", line 26, in request_context_middleware
    return await handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/http/ban.py", line 85, in ban_middleware
    return await handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/http/auth.py", line 242, in auth_middleware
    return await handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/http/headers.py", line 32, in headers_middleware
    response = await handler(request)
               ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/helpers/http.py", line 73, in handle
    result = await handler(request, **request.match_info)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/http/data_validator.py", line 70, in wrapper
    return await method(view, request, data, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/smart_irrigation/websockets.py", line 197, in post
    await coordinator.async_update_zone_config(zone, data)
  File "/config/custom_components/smart_irrigation/__init__.py", line 1355, in async_update_zone_config
    await self._async_calculate_all()
  File "/config/custom_components/smart_irrigation/__init__.py", line 807, in _async_calculate_all
    sensor_values = await self.apply_aggregates_to_mapping_data(
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/smart_irrigation/__init__.py", line 734, in apply_aggregates_to_mapping_data
    resultdata[key] = statistics.mean(d)
                      ^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/statistics.py", line 484, in mean
    T, total, n = _sum(data)
                  ^^^^^^^^^^
  File "/usr/local/lib/python3.12/statistics.py", line 193, in _sum
    for n, d in map(_exact_ratio, values):
  File "/usr/local/lib/python3.12/statistics.py", line 330, in _exact_ratio
    raise TypeError(msg)
TypeError: can't convert type 'str' to numerator/denominator
```

### How to reproduce

for me, the error started occuring after updating to version 2024.6.3. updating to 2024.6.6 (see issue https://github.com/jeroenterheerdt/HAsmartirrigation/issues/398#issue-2342270997) did not fix it.

### Which version are you running?

2024.6.6

### Diagnostics file

- [X] I have attached a diagnostics file

### Additional information

[home-assistant_smart_irrigation_2024-06-10T19-00-14.710Z.log](https://github.com/user-attachments/files/15777970/home-assistant_smart_irrigation_2024-06-10T19-00-14.710Z.log)

**Comment by jeroenterheerdt** [comment]
should be fixed in 2024.6.7

**Comment by MrBurnst** [comment]
jup, thanks!
